lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
> Unfortunately, this doesn't seem to work either if user explicitly sets forceId = false in model file, for e.g book.json file. ModelClass.definition.settings.forceId logic seems to look at the base model's (Model) setting rather than the book model's setting.

I think you should be using `ModelClass.settings.forceId` here.

There are two things I am concerned about:

 - We should keep the logic deciding how to treat`forceId: undefined` in a single place.
 - When `forceId` is effectively `false`, then the `id` property should not be marked as `updateOnly`.

I did not realize that `setupDataAccess` is called too late in our setup process. In that case, I am proposing to move all logic determining `forceId` from `setupDataAccess` to this method (`ModelBuilder.prototype.define`). I am talking about the following lines:

```js
if (idProp.generated && forceId !== false) {
  forceId = true;
}
// set the calculated forceId back to settings
settings.forceId = forceId;
```

I believe we should have all required information already available here. The only thing that can change in `setupDataAccess` is the type of the `id` property, but that is not affecting `forceId` (at least I think so).

Thoughts?